internal/refreshtoken/refreshtoken.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package refreshtoken
+
+import (
+	"bytes"
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/errors"
+)
+
+type ResourceType int
+
+const (
+	ResourceTypeUnknown ResourceType = iota
+	ResourceTypeSession
+	ResourceTypeTarget
+)
+
+// GrantsHasher defines the interface used
+// to retrieve a hash of a users grants.
+type GrantsHasher interface {
+	GrantsHash(ctx context.Context) ([]byte, error)
+}
+
+// A RefreshToken is returned in list endpoints for the purposes of pagination
+type RefreshToken struct {
+	CreatedTime         time.Time
+	ResourceType        ResourceType
+	PermissionsHash     []byte
+	LastItemId          string
+	LastItemUpdatedTime time.Time
+}
+
+// ValidateRefreshToken validates a refresh token.
+func ValidateRefreshToken(
```

Comment:
This function is only used in API handlers to validate incoming refresh tokens. Domains can create their own refresh tokens from trusted data using the struct directly.